src/storage/compression/rle.cpp

```diff
@@ -316,9 +316,41 @@ void RLEScanPartial(ColumnSegment &segment, ColumnScanState &state, idx_t scan_c
 	}
 }
 
+static bool CanEmitConstantVector(idx_t position, idx_t run_length, idx_t scan_count) {
+	D_ASSERT(position < run_length);
+	auto remaining_in_run = run_length - position;
+	// The amount of values left in this run are equal or greater than the amount of values we need to scan
+	return remaining_in_run >= scan_count;
+}
+
+template <class T>
+static void RLEScanConstant(ColumnSegment &segment, RLEScanState<T> &scan_state, idx_t scan_count, Vector &result) {
+	auto data = scan_state.handle.Ptr() + segment.GetBlockOffset();
+	auto data_pointer = (T *)(data + RLEConstants::RLE_HEADER_SIZE);
+	auto index_pointer = (rle_count_t *)(data + scan_state.rle_count_offset);
+
+	result.SetVectorType(VectorType::CONSTANT_VECTOR);
+	auto result_data = ConstantVector::GetData<T>(result);
+	result_data[0] = data_pointer[scan_state.entry_pos];
+	scan_state.position_in_entry += scan_count;
+	if (scan_state.position_in_entry >= index_pointer[scan_state.entry_pos]) {
+		// handled all entries in this RLE value
+		// move to the next entry
+		scan_state.entry_pos++;
```

Comment:
Can we add a test where this is triggered? e.g. having the values `0x4096 rows, 1x2048 rows, 2x2048 rows, 3x1024 rows, 4x1024 rows, 5x512 rows, 6x512 rows, 7x512 rows, 8x512 rows`

Reply:
I've added a `vector_type(col: ANY) -> VARCHAR` scalar function that outputs the vector type for the value that's input.

Using that in `rle_constant.test` allows us to determine if the right vector type is output from the scan 👍